lib/SILOptimizer/Mandatory/AddressLowering.cpp

```diff
@@ -1815,8 +1835,18 @@ SILInstruction *AddressMaterialization::getProjectionInsertionPoint(
       retval = storage.storageAddress->getNextInstruction();
       break;
     }
+    if (!obstruction) {
+      // It's only necessary to consider the obstruction to the earliest
+      // projection because obstructions to any others must dominate it.
+      if (auto *loi = pass.getLatestOpeningInst(pair))
+        obstruction = loi->getNextInstruction();
```

Comment:
Same question about obstructions: Why do we only consider the first latest opening instruction?
But this time we need to consider def projections. (Maybe that doesn't matter.)

Reply:
Fixed to consider all obstructions.  Instead of collecting them into a list, it relies on the fact that the opening instructions all must either dominate or be dominated by one another.